app/src/main/java/com/example/finding_tory/ItemViewActivity.java

```diff
@@ -145,7 +148,18 @@ private void setItemView(Item passedItem) {
         TextView item_serial_number = findViewById(R.id.item_serial_number_text);
         item_serial_number.setText(passedItem.getSerialNumber());
 
-        TextView item_tags = findViewById(R.id.item_tags);
-        item_tags.setText(passedItem.getTagsString());
+        LinearLayout tagsContainer = findViewById(R.id.item_tag_container);
+        tagsContainer.removeAllViews();
+        LayoutInflater inflater = LayoutInflater.from(this);
+        for (String tag : passedItem.getItemTags()) {
+            View tagView = inflater.inflate(R.layout.tag_item_layout, tagsContainer, false);
+            TextView tagTextView = tagView.findViewById(R.id.tag_text);
+            tagTextView.setText(tag);
+
+            ImageButton removeTagButton = tagView.findViewById(R.id.remove_tag_button);
+            removeTagButton.setVisibility(View.GONE);
+            tagsContainer.addView(tagView); // Add the tag view to the container
+
+        }
     }
-}
+}
```

Comment:
File should end in newline
```suggestion
}

```